server/src/main/java/org/elasticsearch/search/aggregations/InternalOrder.java

```diff
@@ -185,25 +188,27 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                     .map(oe -> oe.partiallyBuiltBucketComparator(ordinalReader, aggregator))
                     .collect(toList());
             return (lhs, rhs) -> {
-                Iterator<Comparator<T>> itr = comparators.iterator();
-                int result;
-                do {
-                    result = itr.next().compare(lhs, rhs);
-                } while (result == 0 && itr.hasNext());
-                return result;
+                for (Comparator<T> c : comparators) {
+                    int result = c.compare(lhs, rhs);
+                    if (result != 0) {
+                        return result;
+                    }
+                }
+                return 0;
```

Comment:
Now that we're sure comparators will never be empty, do you think it is worth making these two changes?

Reply:
These changes aren't necessary, yes. I would keep them because (a) the new code is about the same complexity as the old one and (b) in case the nonempty restriction gets dropped in the future you don't need to remember to update these places.

But, it's a minor thing, so I'd say it's up to you if you want it or not.

Reply:
Yeah, I think this fine, sure. I'll keep it!